File: iup2/test/lua/timer.lua
local timer = iup.timer{
  time = 5,
  action_cb=function(self)
--self.run = "NO"
    print("timer called"..false)
  end,
}


local btstart = iup.button{
  title="start",
  action=function()
    print("timer started")
    timer.run = "YES"
  end
}


local btstop = iup.button{
  title="stop",
  action=function()
    print("timer stopped")
    timer.run = "NO"
  end
}


local dlg = iup.dialog{
  iup.hbox{
    btstart,
    btstop,
  };
  title = "timer bug"
}
dlg:popup(iup.CENTER,iup.CENTER) 

File: iup/test/timer.c
#include <stdio.h>
#include "iup.h"

Ihandle *timer1, *timer2, *timer3;

int timer_cb(Ihandle *ih)
{
  if (ih == timer1)
    printf("timer 1 called\n");

  if (ih == timer2)
  {
    printf("timer 2 called and stopped\n");
    IupSetAttribute(ih, "RUN", "NO");
  }

  if (ih == timer3)
  {
    printf("timer 3 called and CLOSE returned.\n");
    return IUP_CLOSE;
  }

  return IUP_DEFAULT;
}

int main(int argc, char* argv[])
{
  Ihandle *dlg;

  IupOpen(&argc, &argv);

  dlg = IupDialog(NULL);
  IupSetAttribute(dlg, "TITLE", "IupTimer Test");
  IupSetAttribute(dlg, "SIZE", "200x100");

  IupShow(dlg);

  timer1 = IupTimer();
  timer2 = IupTimer();
  timer3 = IupTimer();

  IupSetAttribute(timer1, "TIME",  "100");
  IupSetAttribute(timer1, "RUN",   "YES");
  IupSetCallback(timer1, "ACTION_CB", (Icallback)timer_cb);

  IupSetAttribute(timer2, "TIME",  "400");
  IupSetAttribute(timer2, "RUN",   "YES");
  IupSetCallback(timer2, "ACTION_CB", (Icallback)timer_cb);

  IupSetAttribute(timer3, "TIME",  "5000");
  IupSetAttribute(timer3, "RUN",   "YES");
  IupSetCallback(timer3, "ACTION_CB", (Icallback)timer_cb);

  IupMainLoop();

  IupDestroy(dlg);
  IupDestroy(timer1);
  IupDestroy(timer2);
  IupDestroy(timer3);

  IupClose();

  return 0;
}


